### PR TITLE
Feature/sg 000 mapillary dataset

### PR DESCRIPTION
**Pull request by NatanBagrov**

### CODE REVIEW

**Review by lkdci — COMMENTED** [review]
Thanks, few comments bellow

File: src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+train_dataloader_params:
+  shuffle: True
+  batch_size: 4
+  num_workers: 8
+  drop_last: True
+  pin_memory: True
+
+val_dataloader_params:
+  batch_size: 1
```
[the remark this answers is not in the document]
Comment (NatanBagrov) [reply src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml:65]: Thanks, taken care in [5035a37](https://github.com/Deci-AI/super-gradients/pull/594/commits/5035a37b713285248de8a6a9466c740ac7a8209e)

File: src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  transforms:
+    - SegRescale:
+        long_size: 2048
+
+    - SegPadToDivisible:
+        divisible_value: 32
+        fill_mask: ${dataset_params.fill_value}
+
+    - SegCropImageAndMask:
+        crop_size: 512
```
[the remark this answers is not in the document]
Comment (NatanBagrov) [reply src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml]: Thanks, taken care in [5035a37](https://github.com/Deci-AI/super-gradients/pull/594/commits/5035a37b713285248de8a6a9466c740ac7a8209e)

File: src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+config_version: '1.2'
+
+fill_values:
```
[the remark this answers is not in the document]
Comment (NatanBagrov) [reply src/super_gradients/recipes/dataset_params/mapillary_dataset_params.yaml]: Thanks, taken care in [5035a37](https://github.com/Deci-AI/super-gradients/pull/594/commits/5035a37b713285248de8a6a9466c740ac7a8209e). WDYT?

**Review by lkdci — DISMISSED** [review]
LGTM

File: src/super_gradients/training/transforms/transforms.py [hunk 4]
```diff
@@ -320,6 +320,41 @@ def check_valid_arguments(self):
...
+
+        padded_w = int(math.ceil(w / self.divisible_value) * self.divisible_value)
+        padded_h = int(math.ceil(h / self.divisible_value) * self.divisible_value)
+
+        # pad images from center symmetrically
+        if w != padded_w or h != padded_h:
+            padh = padded_h - h
+            padw = padded_w - w
+
+            image = ImageOps.expand(image, border=(0, 0, padw, padh), fill=self.fill_image)
```
Comment (BloodAxe) [inline src/super_gradients/training/transforms/transforms.py:346]: The comment says `pad images from center symmetrically`, while ImageOps.expand uses pad on right & bottom.

File: src/super_gradients/training/datasets/segmentation_datasets/mapillary_dataset.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+        self.ignore_label = self.label_names.index("Unlabeled")
+        # SG format requires returning classes as label names without ignore labels, it is also often used to calculate
+        # the num of classes.
+        self.classes = self.label_names[:-1]
+
+    def _generate_samples_and_targets(self):
+        samples_dir = os.path.join(self.root, self.samples_sub_directory)
+        labels_dir = os.path.join(self.root, self.targets_sub_directory)
+
+        sample_names = [n for n in os.listdir(samples_dir) if n.endswith(self.sample_extension)]
```
Comment (BloodAxe) [inline src/super_gradients/training/datasets/segmentation_datasets/mapillary_dataset.py:90]: `os.listdir` does not guarantee that order of files would be the same in different OS. 
`sorted(os.listdir)` would solve this problem.

File: src/super_gradients/common/environment/omegaconf_utils.py [hunk 6]
```diff
@@ -36,3 +36,4 @@ def register_hydra_resolvers():
     OmegaConf.register_new_resolver("getitem", lambda container, key: container[key], replace=True)  # get item from a container (list, dict...)
     OmegaConf.register_new_resolver("first", lambda lst: lst[0], replace=True)  # get the first item from a list
     OmegaConf.register_new_resolver("last", lambda lst: lst[-1], replace=True)  # get the last item from a list
+    OmegaConf.register_new_resolver("choose", lambda key, options: options[key], replace=True)
```
Comment (BloodAxe) [inline src/super_gradients/common/environment/omegaconf_utils.py:39]: What is the difference of this resolver to getitem? The order of container/key is swapped but apart from this they are the same

Comment (NatanBagrov) [reply src/super_gradients/common/environment/omegaconf_utils.py]: Apparently, I missed it.
[77ac604](https://github.com/Deci-AI/super-gradients/pull/594/commits/77ac60477c40fd249040c61a185d8fcb49c6e25d)

File: src/super_gradients/training/datasets/segmentation_datasets/mapillary_dataset.py (hunk 5 above)
Comment (NatanBagrov) [reply src/super_gradients/training/datasets/segmentation_datasets/mapillary_dataset.py]: This shouldn't matter as long as results are consistent between two calls.
Anyway, your repo your rules - [d1b37d2](https://github.com/Deci-AI/super-gradients/pull/594/commits/d1b37d2ff00525655b0de31b1eeed09af4724a2f)

File: src/super_gradients/training/transforms/transforms.py (hunk 4 above)
Comment (NatanBagrov) [reply src/super_gradients/training/transforms/transforms.py:345]: Thanks, and sorry - [d3df33d](https://github.com/Deci-AI/super-gradients/pull/594/commits/d3df33d1126bd220fa6efa87b1fc3c0edc311f4c)

**Comment by NatanBagrov** [comment]
Thanks @BloodAxe for the comments, addressed.

**Comment by BloodAxe** [comment]
LGTM

**Review by Louis-Dupont — COMMENTED** [review]
Looks great, but maybe just add a small test to make sure that the "get" function properly instantiates the dataset?

You can duplicate this code and just change according to your dataset
https://github.com/Deci-AI/super-gradients/blob/69d8d19813964022af192a34b6e7853edac34a75/tests/unit_tests/datalaoder_factory_test.py#L93-L102

**Review by Louis-Dupont — APPROVED** [review]
LGTM

**Comment by NatanBagrov** [comment]
Please, dear committee, @shaydeci @ofrimasad  approve my PR.
I already got an approval from @lkdci , @BloodAxe , @Louis-Dupont .

**Review by ofrimasad — APPROVED** [review]
LGTM

**Pull request merged**